src/download.rs

```diff
@@ -480,9 +501,26 @@ fuels = { version = "0.1", features = ["some-feature"] }
             body,
         );
         let res = s.parse::<Response>().unwrap();
-        assert!(write_response_with_progress_bar(res, &mut data).is_ok());
+        assert!(write_response_with_progress_bar(res, &mut data, String::new()).is_ok());
         let written_res = String::from_utf8(data)?;
         assert!(written_res.trim().eq(&body));
         Ok(())
     }
+
+    #[test]
+    #[should_panic]
+    fn test_write_response_with_progress_bar_throw_error() {
+        let mut mock_writer = MockWriter;
+        let len = 9000;
+        let body = "A".repeat(len);
+        let s = format!(
+            "HTTP/1.1 200 OK\r\n\
+                 Content-Length: {}\r\n
+                 \r\n
+                 {}",
+            len, body,
+        );
+        let res = s.parse::<Response>().unwrap();
+        assert!(write_response_with_progress_bar(res, &mut mock_writer, String::new()).is_ok());
```

Comment:
I don't believe that we need to assert or `is_ok()` here, no?
If the call fails then the panic will be triggered and the test will fail as expected, or am I misunderstanding?

Reply:
`anyhow::bail!` doestn't triigger panic
![image](https://github.com/FuelLabs/fuelup/assets/25278203/9a88945d-8091-46da-8e5e-749533f0593c)

Reply:
Or we can remove [should_panic], just use assert
![image](https://github.com/FuelLabs/fuelup/assets/25278203/0ffe7fdf-6084-47ad-b77d-0ce73e9d9484)

Reply:
I see. If we are able to catch the exact error and assert that the specific error matches what we expect in the test then we should do that.

Reply:
> I see. If we are able to catch the exact error and assert that the specific error matches what we expect in the test then we should do that.

Done